fix: perps trading OK-42995 OK-43050 OK-43052 OK-43083 OK-43183 OK-43185 OK-43191 OK-433192 OK-43196 OK-43218 OK-43184 OK-43239 OK-43397 OK-43398 , feat: enhance perps trading system with comprehensive improvements

Minnzen opened the pull request:
- Add order confirmation hook and improve trading flow
- Enhance margin mode selector with better validation
- Improve liquidation price display and calculations
- Refactor trading panel components for better modularity
- Add new order management features and UI improvements
- Update service layer with enhanced error handling
- Consolidate utility functions and improve type safety

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Added “Skip order confirmation” setting with a persistent “Don’t show this again” option.
  - Bulk-cancel all open orders from the orders list header.

- Improvements
  - Consistent price formatting to significant digits and smarter market-price autofill.
  - Faster, more reliable order submission and symbol switching; TP/SL pricing and slippage handling improved.
  - Margin mode shows a loading spinner during updates.
  - Deposit/Withdraw shows balances with two decimals and tappable MAX.
  - Liquidation price shows “N/A” when unavailable.

- Bug Fixes
  - Slider now correctly accepts 0 as a valid value.
  - Market vs. limit order timing and price handling corrected (IOC for market, GTC for limit).
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Minnzen committed:
feat: enhance perps trading system with comprehensive improvements

- Add order confirmation hook and improve trading flow
- Enhance margin mode selector with better validation
- Improve liquidation price display and calculations
- Refactor trading panel components for better modularity
- Add new order management features and UI improvements
- Update service layer with enhanced error handling
- Consolidate utility functions and improve type safety

revan-zhang commented:
### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/c59f2925-9c9c-4cda-8177-f1a504c32a9f?refs=75a62a33661e954726feb7d3f034d3e5088d39bb&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/c59f2925-9c9c-4cda-8177-f1a504c32a9f/license?refs=75a62a33661e954726feb7d3f034d3e5088d39bb&source=prChecksComment)

Minnzen committed:
Merge branch 'x' into fix/perps-trading

originalix commented on packages/kit-bg/src/dbs/simple/entity/SimpleDbEntityPerp.ts:
```diff
@@ -28,8 +29,8 @@ export interface ISimpleDbPerpData {
   hyperliquidCurrentToken?: string;
-  tradingUniverse: IPerpsUniverse[] | undefined;
-  marginTables: IMarginTables | undefined;
+  tradingUniverse?: IPerpsUniverse[] | undefined;
```
整体排查下 this.setPerpData 的调用位置，有关这两个必填字段赋值的地方可以移除了

originalix commented on packages/kit/src/states/jotai/contexts/hyperliquid/index.ts:
```diff
@@ -9,7 +9,7 @@ export {
 } from './atoms';
 
-export { useCurrentTokenAtom, useSubscriptionActiveAtom } from './atoms';
+export { usePerpsCurrentTokenAtom, useSubscriptionActiveAtom } from './atoms';
```
逐渐替换 usePerpsSelectedSymbolAtom

kit 层的可以慢慢废弃了

Minnzen replied:
ok

originalix commented on packages/kit/src/views/Perp/components/TokenSelector/PerpTokenSelector.tsx:
```diff
@@ -166,7 +166,7 @@ const PerpTokenSelectorContentMemo = memo(PerpTokenSelectorContent);
   const themeVariant = useThemeVariant();
   const [isOpen, setIsOpen] = useState(false);
-  const [currentToken] = useCurrentTokenAtom();
+  const [currentToken] = usePerpsCurrentTokenAtom();
```
usePerpsSelectedSymbolAtom

originalix commented on packages/shared/src/utils/perpsUtils.ts:
```diff
@@ -464,26 +467,75 @@ function validatePriceInput(input: string, szDecimals = 2): boolean {
-  if (!price || Number.isNaN(price)) return '0';
+  if (!price) return '0';
+
+  const numPrice = Number(price);
+  if (Number.isNaN(numPrice)) return '0';
+
+  const priceStr =
+    numPrice < 1e-6 || numPrice > 1e12
+      ? numPrice.toFixed(20).replace(/\.?0+$/, '')
+      : numPrice.toString();
```
bigNumber

Minnzen committed:
refactor: optimize perps data flow and remove redundant hooks

- Remove useHyperliquid hook and consolidate data access patterns
- Update hooks to use perpsSelectedSymbolAtom for consistent token selection
- Improve usePerpMarketData with better data flow management
- Clean up unused imports and simplify component dependencies
- Optimize perpsUtils calculation functions

Minnzen committed:
Merge branch 'x' into fix/perps-trading

Minnzen committed:
refactor: simplify builder fee handling in Hyperliquid services

- Remove redundant getBuilder method from ServiceHyperliquid
- Consolidate builder fee usage in ServiceHyperliquidExchange
- Use direct _builderFeeInfo reference instead of method calls

Minnzen committed:
fix: lint

Minnzen commented:
@coderabbitai reslove

Minnzen commented:
@coderabbitai resolve

Minnzen committed:
Merge branch 'x' into fix/perps-trading

Minnzen committed:
Merge branch 'x' into fix/perps-trading

Minnzen commented:
@coderabbitai resolve
